cmd/config.go

```diff
@@ -554,27 +555,29 @@ type DNSProvider struct {
 	SRVLookup ServiceDomain `validate:"required"`
 }
 
-// HMACKeyConfig specifies a path to a file containing an HMAC key. The key must
-// consist of 256 bits of random data to be suitable for use as a 256-bit
-// hashing key (e.g., the output of `openssl rand -hex 32`).
+// HMACKeyConfig specifies a path to a file containing a hexadecimal-encoded
+// HMAC key. The key must represent exactly 256 bits (32 bytes) of random data
+// to be suitable for use as a 256-bit hashing key (e.g., the output of `openssl
+// rand -hex 32`).
 type HMACKeyConfig struct {
 	KeyFile string `validate:"required"`
 }
 
-// Load loads the HMAC key from the file, ensures it is exactly 32 characters
-// in length, and returns it as a byte slice.
+// Load reads the HMAC key from the file, decodes it from hexadecimal, ensures
+// it represents exactly 256 bits (32 bytes), and returns it as a byte slice.
 func (hc *HMACKeyConfig) Load() ([]byte, error) {
 	contents, err := os.ReadFile(hc.KeyFile)
 	if err != nil {
 		return nil, err
 	}
-	trimmed := strings.TrimRight(string(contents), "\n")
 
-	if len(trimmed) != 32 {
-		return nil, fmt.Errorf(
-			"validating HMAC key, length must be 32 alphanumeric characters, got %d",
-			len(trimmed),
-		)
+	decoded, err := hex.DecodeString(strings.TrimSpace(string(contents)))
+	if err != nil {
+		return nil, fmt.Errorf("invalid hexadecimal encoding: %w", err)
+	}
+
+	if len(decoded) != 32 {
+		return nil, errors.New("validating HMAC key, must be exactly 256 bits (32 bytes) after decoding")
```

Comment:
Looks like this error message has dropped the %d for the actually-received length

Reply:
I'm happy to restore it, but I felt this information was unlikely to be used since we explicitly specify the input the operator should provide.